### PR TITLE
DL-1900 Play 2.6 Upgrade for ACM

### PR DESCRIPTION
**Pull request by jmountain-hmrc**
# DL-1900 Play 2.6 Upgrade for ACM

**Play 2.6 Upgrade**

* Upgraded agent-client-mandate to use Play 2.6. 
* Added basic integration tests and integration test helpers for ACM 

## Checklist

 - [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [x]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [x]  I've executed the acceptance test pack locally to ensure there are no functional regressions
 - [x]  I've added my code using logical, atomic commits, squashing as appropriate - including the JIRA issue number in the commit message
 - [x]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

**Comment by paulanderson1234** [comment]
Unit and Integration tests, Acceptance tests all run successfully locally. Requires code review before merging

File: app/uk/gov/hmrc/agentclientmandate/models/mandateModels.scala [hunk 1]
```diff
@@ -68,7 +68,36 @@ object Status extends Enumeration {
 case class MandateStatus(status: Status, timestamp: DateTime, updatedBy: String)
 
 object MandateStatus {
-  implicit val formats = Json.format[MandateStatus]
+  val statusWrites: Writes[MandateStatus] = new Writes[MandateStatus] {
```
Comment (Ufuoma-Daniel) [inline app/uk/gov/hmrc/agentclientmandate/models/mandateModels.scala]: Why did you need to create a reads and writes for this?

Comment (jmountain-hmrc) [reply app/uk/gov/hmrc/agentclientmandate/models/mandateModels.scala]: The timestamp format needs to preserved for two reasons, as it was converting the timestamp format to a JodaTime date format which wasn't readable. I've enforced millisecond (epoch) format to:

* Preserve in-flight users with saving/reading their timestamp to mongo
* Allow the frontend to read the timestamp. It would not read it at all and caused acceptance tests to fail

File: app/uk/gov/hmrc/agentclientmandate/services/AgentDetailsService.scala [hunk 2]
```diff
@@ -16,26 +16,28 @@
...
+                                           val authConnector: AuthorityConnector,
+                                           val mandateFetchService: MandateFetchService) extends AgentDetailsService
 
 trait AgentDetailsService {
 
   def etmpConnector: EtmpConnector
-
-  def authConnector: AuthConnector
-
+  def authConnector: AuthorityConnector
```
Comment (Ufuoma-Daniel) [inline app/uk/gov/hmrc/agentclientmandate/services/AgentDetailsService.scala]: Did you swap from standard auth to a custom one?

Comment (Ufuoma-Daniel) [reply app/uk/gov/hmrc/agentclientmandate/services/AgentDetailsService.scala]: Wait never mind, you just renamed it. It's a different call from the standard "http.POST to "$serviceUrl/auth/authorise" anyways.

File: it/controllers/MandateCreateServiceISpec.scala [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import helpers.IntegrationSpec
+import play.api.libs.json.Json
+import play.api.libs.ws.WSResponse
+import uk.gov.hmrc.agentclientmandate.models.{CreateMandateDto, Status}
+import uk.gov.hmrc.agentclientmandate.repositories.{MandateFetchStatus, MandateFetched}
+
+class MandateCreateServiceISpec extends IntegrationSpec {
+
+  "agent/[agentcode]/mandate" should {
+    "create a agent client mandate" when {
```
Comment (Ufuoma-Daniel) [inline it/controllers/MandateCreateServiceISpec.scala]: I admire the integration test <3 but why only one?
